src/ws/cockpit-desktop.in

```diff
@@ -153,7 +161,7 @@ esac
 detect_browser
 
 # start the bridge; this needs to run in the normal user session/namespace
-coproc cockpit-bridge
+coproc ${2:+ssh "$2"} cockpit-bridge
```

Comment:
Oh, that was easy! :-)

Reply:
Heh yes, this is essentially a "50 lines of documentation and 50 lines of test case for a ten-character feature implementation :grin:

Reply:
What happens when ssh prompts for a password?  (I guess I should just try.) The perfect implementation would show a login dialog in the browser, no?

Reply:
It will just prompt on the shell, and the browser window will be empty until then:
```
❱❱❱ ./cockpit-desktop / admin@c
Warning: Permanently added '[127.0.0.2]:2201' (ECDSA) to the list of known hosts.
admin@127.0.0.2's password: 
```
This is of course not ideal, and doesn't work at all if you would add an ssh host to a .desktop file. That's why it is an experimental/demo feature. :-) Honestly it's more like a teaching tool how you can plug together cockpit's components rather than a feature that a lot of people would actually use.

Reply:
What if we use `cockpit-ssh` instead of ssh here?  Wouldn't that maybe even magically work?

Reply:
It would certainly be more elegant, but it's not that simple -- there's always at least one challenge, so you need to intercept the first one or two conversation rounds. Also, doing things remotely involves changing the channel messages to include a `host:` property. If you have an idea how to do that, I'm all ears of course!

However, I think there's a certain appeal to using standard ssh.